temba/channels/models.py

```diff
@@ -1188,6 +1207,62 @@ def success(cls, channel, msg, msg_status, start, request_method=None, request_u
                                   request_time=request_time)
 
     @classmethod
+    def send_fcm_message(cls, channel, msg, text):
+        from temba.msgs.models import WIRED
+        start = time.time()
+
+        url = 'https://fcm.googleapis.com/fcm/send'
+        title = channel.config.get(Channel.CONFIG_FCM_TITLE)
+        data = {
+            'data': {
+                'type': 'rapidpro',
+                'title': title,
+                'message': text,
+                'message_id': msg.id
+            },
+            'content_available': False,
+            'to': msg.auth_token,
+            'priority': 'high'
+        }
+
+        if channel.config.get(Channel.CONFIG_FCM_NOTIFICATION):
+            data['notification'] = {
+                'title': title,
+                'body': text
+            }
+            data['content_available'] = True
+
+        payload = json.dumps(data)
+        headers = {'Content-Type': 'application/json',
+                   'Authorization': 'key=%s' % channel.config.get(Channel.CONFIG_FCM_KEY)}
+        headers.update(TEMBA_HEADERS)
+
+        try:
+            response = requests.post(url, data=payload, headers=headers, timeout=5)
+            result = json.loads(response.content) if response.status_code == 200 else None
+        except Exception as e:  # pragma: no cover
+            raise SendException(unicode(e),
+                                method='POST',
+                                url=url,
+                                request=payload,
+                                response="",
+                                response_status=503,
+                                start=start)
+
+        if result and 'success' in result and result.get('success') == 1:
+            external_id = result.get('multicast_id')
+            Channel.success(channel, msg, WIRED, start, 'POST', url, payload, response, external_id)
+        else:
+            raise SendException("Got non-200 response [%d] from Firebase Cloud Messaging" %
+                                response.status_code,
+                                method='POST',
+                                url=url,
+                                request=payload,
+                                response=result,
```

Comment:
response here should probably be `response.content` in case the status code isn't 200 so we can get the body regardless